.drone.yml

```diff
@@ -1943,48 +1943,58 @@ clone:
 
 steps:
   - name: Set up exec runner storage
+    environment:
+      WORKSPACE_DIR: /tmp/build-darwin-amd64
     commands:
-      - mkdir -p /tmp/build-darwin-amd64
-      - chmod -R u+rw /tmp/build-darwin-amd64
-      - rm -rf /tmp/build-darwin-amd64/go
+      - mkdir -p $WORKSPACE_DIR
+      - chmod -R u+rw $WORKSPACE_DIR
+      - rm -rf $WORKSPACE_DIR/go $WORKSPACE_DIR/.ssh
 
   - name: Check out code
     environment:
       GITHUB_PRIVATE_KEY:
         from_secret: GITHUB_PRIVATE_KEY
+      WORKSPACE_DIR: /tmp/build-darwin-amd64
     commands:
-      - mkdir -p /tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport
-      - cd /tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport
+      - mkdir -p $WORKSPACE_DIR/go/src/github.com/gravitational/teleport
+      - cd $WORKSPACE_DIR/go/src/github.com/gravitational/teleport
       - git clone https://github.com/gravitational/teleport.git .
       - git checkout ${DRONE_TAG:-$DRONE_COMMIT}
       # fetch enterprise submodules
-      - mkdir -m 0700 ~/.ssh && echo -n "$GITHUB_PRIVATE_KEY" > ~/.ssh/id_rsa && chmod 600 ~/.ssh/id_rsa
-      - ssh-keyscan -H github.com > ~/.ssh/known_hosts 2>/dev/null && chmod 600 ~/.ssh/known_hosts
-      - git submodule update --init e
+      # suppressing the newline on the end of the private key makes git operations fail on MacOS
+      # with an error like 'Load key "/path/.ssh/id_rsa": invalid format'
+      - mkdir -m 0700 $WORKSPACE_DIR/.ssh && echo "$GITHUB_PRIVATE_KEY" > $WORKSPACE_DIR/.ssh/id_rsa && chmod 600 $WORKSPACE_DIR/.ssh/id_rsa
+      - ssh-keyscan -H github.com > $WORKSPACE_DIR/.ssh/known_hosts 2>/dev/null
+      - chmod 600 $WORKSPACE_DIR/.ssh/known_hosts
+      - GIT_SSH_COMMAND='ssh -i $WORKSPACE_DIR/.ssh/id_rsa -o UserKnownHostsFile=$WORKSPACE_DIR/.ssh/known_hosts -F /dev/null' git submodule update --init e
```

Comment:
`export` this env var once before running any git commands

Reply:
Sorry, I meant `GIT_SSH_COMMAND`, not `WORKSPACE_DIR`

Reply:
Passing it this way makes `GIT_SSH_COMMAND` a shell variable which is passed to the shell running `git submodule update --init e` but nowhere else, right? As we only need to use it twice I didn't see the harm in repeating the shell variable and being explicit about that. AFAIK there's no need for `export` as we don't use it anywhere outside of these two commands.

Reply:
I'm going to remove the `export WORKSPACE_DIR` as it's redundant and will just confuse people in future.